### PR TITLE
Added Iterable to GlyphSet

### PR DESCRIPTION
**Pull request by Pleket**
## Description
I made GlyphSet iterable, to loop over glyphs in a for-each loop. 

## Motivation and Context
Right now, it is only possible to manually loop through the glyphs. This abstracts away from the underlying data structure. 

## How Has This Been Tested?
So far, there are no test cases created for glyphset, and as such I did not add a new test file either. If this is desired, the entire class will likely need to be tested, which is probably better left for the creator of GlyphSet. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I did `npm run test` and all tests passed green (including code styling checks).
- [ ] I have added tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the **README** accordingly.
- [x] I have read the **CONTRIBUTING** document.

This is my first ever contribution to an open-source project, so don't hesitate to criticize everything in this pull request; JavaScript is not my main language.

### CODE REVIEW

**Comment by Connum** [comment]
I think putting tests for this into test/font.js would be fine for now!

**Comment by Pleket** [comment]
Oddly enough, my lint wasn't giving me these errors. I'm fixing it right now.

**Comment by Pleket** [comment]
I fixed my iterator and added test cases which all pass, but the linter goes a bit wild so I hope I'm not missing any obvious semicolons. 

If it's not desired to add some kind of filtering for undefined glyph indices then I would consider it finished!

**Comment by Pleket** [comment]
Would love to have this reviewed!

File: test/font.js [hunk 1]
```diff
@@ -93,3 +93,51 @@ describe('font.js', function() {
...
+            glyphs: glyphs
+        });
+    });
+
+    describe('GlyphSet iterable', function() {
+        it('must be iterable', function() {
+            assert.ok(glyphs[Symbol.iterator]);
+        });
+
+        it('must iterate over glyphs', function() {
```
Comment (yne) [inline test/font.js:129]: what are you trying to test here ? I don't see any usage of font.glyphs so you are just comparing your `glyphs` array to ... itself ?

File: test/font.js [hunk 2]
```diff
@@ -93,3 +93,51 @@ describe('font.js', function() {
...
+            for (const glyph of glyphs) {
+                assert.ok(glyph instanceof Glyph);
+            }
+        });
+
+        it('must iterate over glyphs in order', function() {
```
Comment (yne) [inline test/font.js:135]: same here, no usage of the OpenType font. Just testing your own declaration.

**Review by yne — CHANGES_REQUESTED** [review]
The tests are not testing your changes.

**Comment by Connum** [comment]
Hi @Pleket, could you please have a look at the requested changes?

File: test/font.js (hunk 1 above)
Comment (Connum) [reply test/font.js]: resolved by changing the test to explicitly test if it is a function

File: test/font.js (hunk 2 above)
Comment (Connum) [reply test/font.js:217]: resolved by actually using the GlyphSet instance of the font

**Comment by Connum** [comment]
merged current master, fixed and unified tests, fixed the `this` context of the iterator's `next()` to make it actually work. Please re-review

**Review by yne — APPROVED** [review]
new tests looks better now

**Pull request merged**